processor/cumulativetodeltaprocessor/processor.go

```diff
@@ -68,8 +68,8 @@ func (ctdp *cumulativeToDeltaProcessor) processMetrics(_ context.Context, md pda
 							fromDataPoint := dataPoints.At(l)
 							labelMap := make(map[string]string)
 
-							fromDataPoint.LabelsMap().Range(func(k string, v string) bool {
-								labelMap[k] = v
+							fromDataPoint.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+								labelMap[k] = v.StringVal()
```

Comment:
FYI: Not always the labels are string. Please see https://github.com/open-telemetry/opentelemetry-collector/issues/3815

Reply:
What I mean is you need to followup and do the correct thing, which is to persist the type :)

Reply:
I thought metric labels which are now attribtes can only be string. It would be great if can share some existing PR which did the similar transition. Did you just mean keep them as pdata.AttributeValue instead of changing to string?

Reply:
we assume all the metric labels are string type today so we use the concatenated values as key on the same metrics for delta calculation.  I think the delta calculation will be failed if the label type is not `string` in code [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/aws/metrics/metric_calculator.go#L97).

Reply:
Suggest to change https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/aws/metrics/metric_calculator.go#L97 
`kvs = append(kvs, attribute.String(k, v))` 
==>
```kvs = append(kvs, attribute.Any(k, v))```

Reply:
> Suggest to change https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/aws/metrics/metric_calculator.go#L97
> `kvs = append(kvs, attribute.String(k, v))`
> ==>
> `kvs = append(kvs, attribute.Any(k, v))`

Don't do that.  [`attribute.Any()` is deprecated](https://github.com/open-telemetry/opentelemetry-go/pull/2181/files) because it can silently drop data that is not a valid type and generally didn't behave consistently with the other value initializers.

Reply:
I think rather than using `v.StringVal()`, we can use `pdata.AttributeValueToString(v)`. I believe this way we are not changing the type of the existing attribute but getting the string version of an attribute which we can use for generating map key.

Reply:
Let's do a separate PR and discuss there